ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.cache
+
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicLong
+
+import akka.NotUsed
+import akka.stream.scaladsl.{Flow, Sink}
+import com.daml.ledger.participant.state.index.v2.ContractStore
+import com.daml.ledger.participant.state.v1.Offset
+import com.daml.lf.transaction.GlobalKey
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.{Metrics, Timed}
+import com.daml.platform.store.cache.ContractStateValue._
+import com.daml.platform.store.cache.ContractKeyStateValue._
+import com.daml.platform.store.cache.MutableCacheBackedContractStore.{
+  ContractNotFound,
+  EmptyContractIds,
+}
+import com.daml.platform.store.dao.events.ContractStateEvent
+import com.daml.platform.store.dao.events.ContractStateEvent.LedgerEndMarker
+import com.daml.platform.store.interfaces.LedgerDaoContractsReader
+import com.daml.platform.store.interfaces.LedgerDaoContractsReader.{
+  ActiveContract,
+  ArchivedContract,
+  ContractState,
+  KeyState,
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MutableCacheBackedContractStore(
+    metrics: Metrics,
+    contractsReader: LedgerDaoContractsReader,
+    signalNewLedgerHead: Offset => Unit,
```

Comment:
What are the requirements for this function so that a caller knows how to correctly instantiate it?